File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.cpp
class Solution {
public:
    void Addone(string &s){
        int n =s.size();
        int carry =1;
        for(int i=n-1;i>=0;i--){
            if(s[i]=='1' && carry ==1){
                s[i]='0';
                carry=1;

            }
            else if(s[i]=='0' && carry == 1){
                s[i]='1';
                carry =0;
                break;
            }
        }
        if(carry==1){
            s='1'+s;
        }
    }
 
    int numSteps(string s) {
        int step =0;
        while(s!="1")
        {
            if(s.back()=='0'){
                s.pop_back();
            }
            else{
                Addone(s);
            }
            step++;


        }
        return step;
        
    }
};

File: 1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/1404-number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        n =len(s)
        step=0
        carry =0
        for i in range(n-1,0,-1):
            b = int(s[i])+carry
            if b % 2 ==0:
                step+=1
            else:
                step += 2
                carry =1
        return step+carry
        